e2e/support/test_support.go

```diff
@@ -350,6 +352,11 @@ func KamelRunWithID(t *testing.T, ctx context.Context, operatorID string, namesp
 }
 
 func KamelRunWithContext(t *testing.T, ctx context.Context, operatorID string, namespace string, args ...string) *cobra.Command {
+	if runtime.GOARCH == "arm64" && !slices.Contains(args, "builder.platforms") {
+		builderPlatform := "linux/arm64"
+		fmt.Printf("Add trait to run command: -t builder.platforms=%s\n", builderPlatform)
+		args = append(args, "-t", "builder.platforms="+builderPlatform)
```

Comment:
I don't think this is safe. We cannot include such an implicit configuration in all tests. Ideally, if you want to proceed in this direction, you should find a way to include the configuration in the IntegrationPlatform or IntegrationProfile as suggested in https://github.com/apache/camel-k/pull/5403#pullrequestreview-2017276065